src/main/java/seedu/address/logic/commands/DeleteAssignmentCommand.java

```diff
@@ -39,29 +45,58 @@ public class DeleteAssignmentCommand extends Command {
 
     public static final String MESSAGE_NO_STUDENT_FOUND = "No such student found!";
     public static final String MESSAGE_NO_ASSIGNMENT_FOUND = "No such assignment found!";
+    public static final String MESSAGE_DUPLICATE_STUDENT = "There is more than 1 student of the same name \n"
+            + "Their student numbers are as follows: %1$s \n"
+            + MESSAGE_USAGE;
 
     public final AssignmentQuery assignmentQuery;
     public final Name name;
+    public final Optional<StudentNumber> studentNumber;
 
     /**
      * Creates an DeleteAssignmentCommand to add the specified {@code Assignment}
      */
     public DeleteAssignmentCommand(Name name, AssignmentQuery assignmentQuery) {
         this.name = name;
         this.assignmentQuery = assignmentQuery;
+        this.studentNumber = Optional.empty();
+    }
+
+    /**
+     * Creates an DeleteAssignmentCommand to add the specified {@code Assignment}
+     */
+    public DeleteAssignmentCommand(Name name, AssignmentQuery assignmentQuery, StudentNumber studentNumber) {
+        this.name = name;
+        this.assignmentQuery = assignmentQuery;
+        this.studentNumber = Optional.of(studentNumber);
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        Student student = model.getStudentByName(name);
+        List<Student> studentList = model.getAllStudentByName(name);
+
+        if (this.studentNumber.isPresent()) {
+            List<Student> filteredStudentList =
+                    studentList.stream().filter(s -> s.getStudentNumber().equals(studentNumber.get())).toList();
+            if (filteredStudentList.isEmpty()) {
+                throw new CommandException(MESSAGE_NO_STUDENT_FOUND);
+            }
+            Student student = filteredStudentList.get(0);
+            Assignment assignment = student.deleteAssignment(assignmentQuery);
+            return new CommandResult(String.format(MESSAGE_SUCCESS, assignment.getAssignmentName(), student.getName()));
+        }
 
-        if (student == null) {
+        if (studentList.isEmpty()) {
             throw new CommandException(MESSAGE_NO_STUDENT_FOUND);
         }
 
-        Assignment assignment = student.deleteAssignment(assignmentQuery);
+        if (studentList.size() > 1) {
+            throw new CommandException(MESSAGE_DUPLICATE_STUDENT);
```

Comment:
You have not implemented the part where you format the student numbers is it? Cause your String constant contains a placeholder for an argument.

Reply:
Fixed